### ISSUE TITLE
The Android Studio emulator toolbar appears in Alt-Tab

### ISSUE THREAD

**Issue by ShlomoCode** [lwouis/alt-tab-macos#3496]
**Describe the bug**

Android Studio emulators have a sort of vertical toolbar attached:
![CleanShot 2024-07-18 at 06 35 19@2x](https://github.com/user-attachments/assets/448f2851-03e7-4577-a0b3-2d7a4a254b7c)

Alt Tab displays the toolbar as a separate window:
![CleanShot 2024-07-18 at 06 36 18@2x](https://github.com/user-attachments/assets/30a9028a-731b-4ff4-97dd-cbe6ba080b10)

This is NOT a regression by https://github.com/lwouis/alt-tab-macos/commit/2e15732,
I checked with v6.71.0 and the toolbar appears there as well.

If there is any additional information you need, please let me know!

* AltTab version: 6.72.0
* macOS version: 14.5

**Comment by lwouis** [comment]
Hi,

We already have hard-coded a [workaround for this](https://github.com/lwouis/alt-tab-macos/blob/97951e7e8e9a1ab480a53279cf0f08160de28301/src/api-wrappers/AXUIElement.swift#L223). I'm surprised it stopped working. Android Studio must have been updated recently.

This issue is not on AltTab side. This vertical bar **is** a window. That's why AltTab can report it.

I think it would be useful to report this issue to Android Studio support. AltTab was working around the issue, but the root cause remains. Also, maybe the Android Studio team *wants* this bar to be a separate window?

Thank you

**Comment by lwouis** [comment]
AltTab displays Standard Windows and Dialogs. If an app use that to make a UI element, it will show in AltTab. If they use another element like a HUD or custom UI, then AltTab will not display it.

There are many "popups", "popovers", little pieces of UI that exist on your OS which don't show in AltTab. This is because they were implemented using relevant APIs, and they are not standard windows.

Thank you

**Comment by ShlomoCode** [comment]
Yes I checked and the subrole is `AXDialog` (`kAXDialogSubrole`), so it doesn't reach the line
```
androidEmulator(runningApp, title)
```
at all.

In the past sidebar was not `AXDialog`?

**Comment by lwouis** [comment]
I think they may be using [child/parent NSWindow APIs](https://developer.apple.com/documentation/appkit/nswindow#1661753)?

**Comment by ShlomoCode** [comment]
The point is that maybe this is the right behavior on their part, and the problem is with AltTab not behaving like Mission Control and the like?

**Comment by ShlomoCode** [comment]
https://issuetracker.google.com/issues/353915729

**Comment by slkmikhail** [comment]
AltTab has started showing tool windows for the last versions of Idea Intellij, Android Studio even they have view mode "Float".
@lwouis Could you consider to add regex to blacklist? Because the workaround by PR takes a long time.

For instance, everybody can to pick an app in blacklist and then set title(regex) of others window of that app which must not be showed by AltTab.

![2024-12-11_16-30-51](https://github.com/user-attachments/assets/1ea2ebc5-ca30-461c-b89a-9d097e2f80c7)
